contracts/cosmwasm-vm/cw-xcall-ibc-connection/src/contract.rs

```diff
@@ -581,12 +577,14 @@ impl<'a> CwIbcConnection<'a> {
         let channel_id = packet.src.channel_id.clone();
         let channel_config = self.get_channel_config(deps.as_ref().storage, &channel_id)?;
         let nid = channel_config.counterparty_nid;
-        let sn = self.get_outgoing_packet_sn(deps.storage, &channel_id, packet.sequence)?;
-
         let n_message: Message = rlp::decode(&packet.data).unwrap();
-        self.remove_outgoing_packet_sn(deps.storage, &channel_id, packet.sequence);
+
+        let Some(sn) = n_message.sn.0 else {
+            return  Ok(Response::new())
+        };
+
```

Comment:
we can do `if n_message.sn.is_none()` as well instead of negating the negative.

Reply:
But I also need the sn, how would i format that optimally?

Reply:
since we have checked it we can just do unwrap where we need it. I have updated it and fixed one rlp test as well.